From a087d86004b11778417ab7de36608d16a8d591bd Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:52 -0400
Subject: [tools] perf/build: Improve the 'stackprotector' feature test

Message-id: <1408462094-14194-184-git-send-email-jolsa@redhat.com>
Patchwork-id: 88186
O-Subject: [PATCH RHEL7.1 BZ1131394 183/325] tools/perf/build: Improve the 'stackprotector' feature test
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 046fa7ae20d2390c65be8eb406efecff15e8b71d
Author: Ingo Molnar <mingo@kernel.org>
Date: Thu Oct 10 09:06:21 2013 +0200

description
===========
Namhyung Kim noticed that the stackprotector testcase was incomplete:

  > The flag being checked should be -"W"stack-protector instead of
  > -"f"stack-protector.  And the gcc manpage says that -Wstack-protector is
  > only active when -fstack-protector is active.  So the end result should
  > look like
  >
  >         $(BUILD) -Werror -fstack-protector -Wstack-protector

Add -Wstack-protector.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index 2eb8346..c70d23e 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -45,7 +45,7 @@ test-stackprotector-all:
  $(BUILD) -Werror -fstack-protector-all
 
 test-stackprotector:
- $(BUILD) -Werror -fstack-protector
+ $(BUILD) -Werror -fstack-protector -Wstack-protector
 
 test-fortify-source:
  $(BUILD) -O2 -Werror -D_FORTIFY_SOURCE=2
-- 
1.7.1